docs/_reference/web-api.md

```diff
@@ -0,0 +1,4601 @@
+<h5 id="paginatemethod-options-shouldstop">paginate(method, options, shouldStop)</h5>
+<strong>Parameters:</strong>
+<table>
+<thead>
+<tr>
+<th align="center">Name</th>
+<th align="center">Type</th>
+<th align="center">Required</th>
+<th></th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td align="center">method</td>
+<td align="center"><code>string</code></td>
+<td align="center">✓</td>
+<td></td>
+</tr>
+<tr>
+<td align="center">options</td>
+<td align="center"><code><a href="#webapicalloptions" title="">WebAPICallOptions</a></code></td>
+<td align="center">✓</td>
+<td></td>
+</tr>
+<tr>
+<td align="center">shouldStop</td>
+<td align="center"><code><a href="#paginatepredicate" title="">PaginatePredicate</a></code></td>
+<td align="center">✓</td>
+<td></td>
+</tr>
+</tbody>
+</table>
+<p><strong>Returns</strong> <code>Promise&#x3C;void></code></p>
+<h5 id="paginatemethod-options-shouldstop-reduce">paginate(method, options, shouldStop, reduce)</h5>
+<strong>Parameters:</strong>
+<table>
+<thead>
+<tr>
+<th align="center">Name</th>
+<th align="center">Type</th>
+<th align="center">Required</th>
+<th></th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td align="center">method</td>
+<td align="center"><code>string</code></td>
+<td align="center">✓</td>
+<td></td>
+</tr>
+<tr>
+<td align="center">options</td>
+<td align="center"><code><a href="#webapicalloptions" title="">WebAPICallOptions</a></code></td>
+<td align="center">✓</td>
+<td></td>
+</tr>
+<tr>
+<td align="center">shouldStop</td>
+<td align="center"><code><a href="#paginatepredicate" title="">PaginatePredicate</a></code></td>
+<td align="center">✓</td>
+<td></td>
+</tr>
+<tr>
+<td align="center">reduce</td>
+<td align="center"><code>PageReducer&#x3C;A></code></td>
+<td align="center">✗</td>
+<td></td>
+</tr>
+</tbody>
+</table>
+<p><strong>Returns</strong> <code>Promise&#x3C;A></code></p>
+<h2 id="functions">Functions</h2>
+<h3 id="addappmetadataopts">addAppMetadata(opts)</h3>
+<p>Appends the app metadata into the User-Agent value</p>
+<strong>Parameters:</strong>
+<table>
+<thead>
+<tr>
+<th align="center">Name</th>
+<th align="center">Type</th>
+<th align="center">Required</th>
+<th></th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td align="center">{ name, version }</td>
+<td align="center"><code>object</code></td>
+<td align="center">✓</td>
+<td></td>
+</tr>
+</tbody>
+</table>
+<p><strong>Returns</strong> <code>void</code></p>
+<h2 id="enums">Enums</h2>
+<h3 id="errorcode">ErrorCode</h3>
+<p>A dictionary of codes for errors produced by this package</p>
+<h4 id="members">Members</h4>
+<ul>
+<li><strong>HTTPError</strong></li>
+<li><strong>PlatformError</strong></li>
+<li><strong>RateLimitedError</strong></li>
+<li><strong>RequestError</strong></li>
+</ul>
+<h3 id="webclientevent">WebClientEvent</h3>
+<h4 id="members-1">Members</h4>
+<ul>
+<li><strong>RATE_LIMITED</strong></li>
+</ul>
+<h2 id="interfaces">Interfaces</h2>
```

Comment:
Future TODO: There should probably be a special case for `@slack/web-api` to not include all these interfaces and methods that represent the Web API methods.